api/envoy/service/auth/v3/external_auth.proto

```diff
@@ -71,6 +72,18 @@ message OkHttpResponse {
   // one if there is a match.
   repeated config.core.v3.HeaderValueOption headers = 2;
 
+  // HTTP entity headers to remove from the original request before dispatching
+  // it to the upstream. This allows the authorization service to act on auth
+  // related headers (like `Authorization`), process them, and consume them.
+  // Under this model, the upstream will either receive the request (if it's
+  // authorized) or not receive it (if it's not), but will not see headers
+  // containing authorization credentials.
+  //
+  // When using the HTTP service this must instead be set by the HTTP
+  // authorization service as a comma separated list like so:
+  // ``x-envoy-authz-headers-to-remove: one-auth-header, another-auth-header``.
```

Comment:
The other case I can see where headers are used is `x-envoy-auth-partial-body`. This is used for Envoy --> upstream, which is different, but it's a little weird to refer to the filter as auth vs. authz.

Reply:
Sure, let me make that consistent.